security-oauth2/src/main/java/io/micronaut/security/oauth2/configuration/OauthClientConfiguration.java

```diff
@@ -83,6 +84,13 @@
      */
     Optional<EndpointConfiguration> getAuthorization();
 
+    /**
+     *
+     * @return The Client Credentials Configuration
+     */
+    @NonNull
+    Optional<ClientCredentialsConfiguration> getClientCredentials();
```

Comment:
This needs a default implementation to not be a breaking change

Reply:
I've added a default implementation which returns an empty Optional.